packages/okta-adapter/src/adapter.ts

```diff
@@ -192,72 +179,75 @@ export default class OktaAdapter implements AdapterOperations {
+      pagination: PAGINATION,
+      fetch: createFetchDefinitions(
+        this.userConfig,
+        shouldAccessPrivateAPIs(this.isOAuthLogin, this.userConfig),
+        getAdminUrl(this.client.baseUrl),
+      ),
+      sources: { openAPI: [OPEN_API_DEFINITIONS] },
+    }
+
+    this.definitions = {
+      ...definitions,
+      fetch: definitionsUtils.mergeWithUserElemIDDefinitions({
+        userElemID: userConfig.fetch.elemID as OktaUserFetchConfig['elemID'],
+        fetchConfig: definitions.fetch,
+      }),
+    }
 
     this.fetchQuery = elementUtils.query.createElementQuery(this.userConfig.fetch, fetchCriteria)
 
     this.paginator = paginator
 
-    const filterContext = {}
     this.createFiltersRunner = usersPromise =>
-      filtersRunner(
+      filterRunner(
         {
-          client,
-          paginator,
+          definitions: this.definitions,
           config: this.userConfig,
           getElemIdFunc,
-          elementsSource,
           fetchQuery: this.fetchQuery,
-          adapterContext: filterContext,
-          adminClient,
+          elementSource: elementsSource,
+          oldApiDefinitions: OLD_API_DEFINITIONS_CONFIG,
           usersPromise,
+          paginator: this.paginator,
+          baseUrl: this.client.baseUrl,
+          isOAuthLogin,
         },
         filterCreators,
         objects.concatObjects,
       )
-    this.fixElementsFunc = combineElementFixers(createFixElementFunctions({ client, config }))
-  }
-
-  @logDuration('generating types from swagger')
-  private async getSwaggerTypes(): Promise<{
-    allTypes: TypeMap
-    parsedConfigs: Record<string, configUtils.RequestableTypeSwaggerConfig>
-  }> {
-    return openapi.generateTypes(OKTA, this.userConfig[API_DEFINITIONS_CONFIG])
+    this.fixElementsFunc = combineElementFixers(createFixElementFunctions({ client, config: this.userConfig }))
   }
 
-  @logDuration('generating instances from service')
-  private async getSwaggerInstances(
-    allTypes: TypeMap,
-    parsedConfigs: Record<string, configUtils.RequestableTypeSwaggerConfig>,
-  ): Promise<fetchUtils.FetchElements<InstanceElement[]>> {
-    const updatedApiDefinitionsConfig = {
-      ...this.userConfig.apiDefinitions,
-      types: {
-        ...parsedConfigs,
-        ..._.mapValues(this.userConfig.apiDefinitions.types, (def, typeName) => ({
-          ...parsedConfigs[typeName],
-          ...def,
-        })),
-      },
+  private async handleClassicEngineOrg(): Promise<definitionsUtils.ConfigChangeSuggestion | undefined> {
+    const { isClassicOrg: isClassicOrgByConfig } = this.userConfig[FETCH_CONFIG]
+    const isClassicOrg = isClassicOrgByConfig ?? (await isClassicEngineOrg(this.client))
+    if (isClassicOrg) {
+      const updatedCustomizations = _.pickBy(
+        this.definitions.fetch.instances.customizations,
+        (_v, key) => !CLASSIC_ENGINE_UNSUPPORTED_TYPES.includes(key),
+      )
```

Comment:
nit: I think this can also be
```suggestion
      const updatedCustomizations = _.omit(
        this.definitions.fetch.instances.customizations,
        CLASSIC_ENGINE_UNSUPPORTED_TYPES,
      )
```